File: problems/1962_remove_stones_to_minimize_the_total/Solution.java
import java.util.*;

class Solution {
    public int minStoneSum(int[] piles, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        for (int p : piles) pq.offer(p);
        for (int i = 0; i < k; i++) {
            int top = pq.poll();
            pq.offer(top - top / 2);
        }
        int sum = 0;
        for (int p : pq) sum += p;
        return sum;
    }
}

File: problems/1962_remove_stones_to_minimize_the_total/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    int minStoneSum(vector<int>& piles, int k) {
        priority_queue<int> pq(piles.begin(), piles.end());
        for (int i = 0; i < k; i++) {
            int top = pq.top(); pq.pop();
            pq.push(top - top / 2);
        }
        int sum = 0;
        while (!pq.empty()) { sum += pq.top(); pq.pop(); }
        return sum;
    }
};
